simplify graph display code`

yamad opened the pull request:
Refactor to reduce duplication in the graph display code of the custom repr.

I'm not sure I like the complex repr string, but this is a first step.

yamad committed:
simplify graph display code`

yamad committed:
fix 3.4 incompatible star args use

pywkm commented on mindgraph/graph.py:
```diff
@@ -51,29 +50,20 @@ def __getitem__(self, key: int) -> "Node":
 
     def __repr__(self) -> str:
-        return self.custom_repr(self, 0)
+        return self.custom_repr(self, depth=0)
```
I would skip here `depth=0`. It's default value, after all.

pywkm commented on mindgraph/graph.py:
```diff
@@ -51,29 +50,20 @@ def __getitem__(self, key: int) -> "Node":
     @staticmethod
     def custom_repr(node: "Node", depth: int = 0) -> str:
+        indent = "    " * depth
+        bullet = "- " if depth != 0 else ""
```
`bullet = "- " if depth else ""`

yamad replied:
This is the one thing I didn't change. I prefer the explicit check against 0 in this case

pywkm replied:
Strange decision.
I was also explicitly checking against 0. There's no other `int` value for which `bool(depth)` would evaluate to `False`.

yamad replied:
Cheers. I'm aware of how the logic works. It's a stylistic choice to make the predicate explicit, which is implicit in your more concise version

pywkm commented on mindgraph/graph.py:
```diff
@@ -51,29 +50,20 @@ def __getitem__(self, key: int) -> "Node":
 
     @staticmethod
     def custom_repr(node: "Node", depth: int = 0) -> str:
+        indent = "    " * depth
```
Please note you are mixing both quote styles (`"`/`'`) in one method. I know it was same before (mixed), and it is mixed in whole project, but it is great opportunity to straighten things up at least in that one place. :) Please chose one.

pywkm commented on mindgraph/graph.py:
```diff
@@ -51,29 +50,20 @@ def __getitem__(self, key: int) -> "Node":
-                                         node.threads))
-
-        return "    " * (depth - 1) + "- {}".format(node.name)
+        children = [node.custom_repr(n, depth+1) for n in node.threads]
```
I'm not sure who and when decided that `custom_rep` should be static, but I don't think it makes sense. 
Maybe it's time to fix it now (modify lines 55-56). In that case, above line would be:
`children = [child.custom_rep(depth + 1) for child in node.threads]`
What do you think?

pywkm commented on mindgraph/graph.py:
```diff
@@ -51,29 +50,20 @@ def __getitem__(self, key: int) -> "Node":
+        bullet = "- " if depth != 0 else ""
 
+        lines = ['{indent}{bullet}{node.name}'.format(**locals())]
         if len(node.threads) > 0:
```
```
if node.threads:
    [...]
```
But maybe something like this would be better: 
afer line 58 add something like `suffix = ":" if node.threads else ""`, and line 60 would then look like: 
`lines = ["{indent}{bullet}{node.name}{suffix}".format(**locals())]`? 
I simply don't think if modifying line[0] afterwards makes sense - we know if there are threads earlier

yamad replied:
good points, i had it with that `suffix` line before but thought the intent may have been clearer if the child handling code was in one place. then... of course, i didn't keep the child handling code in one place anyway. ;-)

pywkm reviewed (commented):
I'm sorry I allowed myself to review. I was also looking for some pull request opportunities (hacktober fest ;)) and was thinking about simplifying repr method. You were quicker, and I overall like your solution, but have some minor remarks.

yamad committed:
further simplify graph display after @pywkm code review

 - a bunch of stylistic changes in response to code review comments
 - more accurate name
 - internal function now is a generator to avoid needless recursive list
 creation

yamad commented:
@pywkm thanks for the review. good points.

sorry for stealing your idea... i'm sure you'll find other bits of style/improvements to make in this file, though!

yamad commented:
for the record, failing the coveralls check in this way is a silly reason to fail a PR.

This PR removed code and therefore the coverage percentage went down, even though the PR doesn't touch any of the non-covered code.

capsulecorplab committed:
Merge branch 'master' into simple_repr

capsulecorplab commented:
@yamad I like the refactor! @pywkm feel free to suggest other improvements :slightly_smiling_face:
